### DISCUSSION TITLE
Conjugate Heat Transfer TO

### DISCUSSION THREAD

**Discussion by sreekarrs** [mdolab/dafoam#393] [General]
Hello,
I have been working with internal cooling channels to be made using additive manufacturing techniques and would like to optimize the geometry.
Is it possible to implement a CHT TO solver in DAFoam?

If yes, can you please direct me on how I could build it?

Thank you
- Sreekar

**Comment by friedenhe** [comment]
We plan to implement the conjugate heat transfer shape optimization to MPhys in the near future. No plans for topology optimization yet but it should be straightforward to extend.

**Reply by sreekarrs** [reply]
Thank you for the prompt response

Can I ask if I can create a CHT TO solver with the DAFoam v3 structure?
If yes, can you please brief me on how I can do that?

Thank you so much for your help

**Comment by friedenhe** [comment]
Yes, you can follow a similar approach as in the aero-structural coupling. This requires significant development... We are trying to get it done sometime this year.

**Reply by girindra1** [reply]
A follow up on the conjugate heat transfer solver. May I ask which solver adaptations from OpenFOAM are available? I can see from the github website they were implemented some time in May 2023 but  I cannot seem to find the information on the documentation website. Thanks for any info.

**Comment by friedenhe** [comment]
Our conjugate heat transfer is implemented differently than OpenFoam's built-in ones. We have not published a paper about it, but we will do it soon. Once published, we will add a new tutorial and documentation about conjugate heat transfer. Please refer to https://github.com/mdolab/dafoam/blob/main/tests/runTests_MphysAeroThermal.py if you want to test it.
